src/server/scripts/Spells/spell_paladin.cpp

```diff
@@ -357,6 +359,41 @@ class spell_pal_blinding_light : public SpellScript
     }
 };
 
+// 1022 - Blessing of Protection
+class spell_pal_blessing_of_protection : public SpellScript
+{
+    PrepareSpellScript(spell_pal_blessing_of_protection);
+    
+    bool Validate(SpellInfo const* /*SpellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_PALADIN_FORBEARANCE, SPELL_PALADIN_IMMUNE_SHIELD_MARKER });
+    }
+    SpellCastResult Forbearance()
+    {
+        if (Unit* caster = GetCaster())
+        {
+            if(caster->HasAura(SPELL_PALADIN_FORBEARANCE) || caster->HasAura(SPELL_PALADIN_IMMUNE_SHIELD_MARKER))
```

Comment:
you don't need to check SPELL_PALADIN_IMMUNE_SHIELD_MARKER it is done by spellsystem